lib/layer/data.py

```diff
@@ -1448,6 +1448,81 @@ def _load_strokemap_from_ora(self, elem, x, y, orazip=None, oradir=None):
+        )
+
+    def get_fillable(self):
+        """True if this layer currently accepts flood fill"""
+        return not self.locked
+
+    ## Flood fill
+
+    def flood_fill(self, x, y, color, bbox, tolerance, dst_layer=None):
+        """Fills a point on the surface with a color
+
+        :param x: Starting point X coordinate
+        :param y: Starting point Y coordinate
+        :param color: an RGB color
+        :type color: tuple
+        :param bbox: Bounding box: limits the fill
+        :type bbox: lib.helpers.Rect or equivalent 4-tuple
+        :param tolerance: how much filled pixels are permitted to vary
+        :type tolerance: float [0.0, 1.0]
+        :param dst_layer: Optional target layer (default is self!)
+        :type dst_layer: PaintingLayer
+
+        The `tolerance` parameter controls how much pixels are permitted to
+        vary from the starting color.  We use the 4D Euclidean distance from
+        the starting point to each pixel under consideration as a metric,
+        scaled so that its range lies between 0.0 and 1.0.
+
+        The default target layer is `self`. This method invalidates the filled
+        area of the target layer's surface, queueing a redraw if it is part of
+        a visible document.
+        """
+        if dst_layer is None:
+            dst_layer = self
+        dst_layer.autosave_dirty = True   # XXX hmm, not working?
+        self._surface.flood_fill(x, y, color, bbox, tolerance,
+                                 dst_surface=dst_layer._surface)
+
+    ## Painting
+
+    def stroke_to(self, brush, x, y, pressure, xtilt, ytilt, dtime, viewzoom, viewrotation):
+        """Render a part of a stroke to the canvas surface
+
+        :param brush: The brush to use for rendering dabs
+        :type brush: lib.brush.Brush
+        :param x: Input event's X coord, translated to document coords
+        :param y: Input event's Y coord, translated to document coords
+        :param pressure: Input event's pressure
+        :param xtilt: Input event's tilt component in the document X direction
+        :param ytilt: Input event's tilt component in the document Y direction
+        :param dtime: Time delta, in seconds
+        :returns: whether the stroke should now be split
+        :rtype: bool
+
+        This method renders zero or more dabs to the surface of this layer,
+        but does not affect the strokemap. Use this for the incremental
+        painting of segments of a stroke sorresponding to single input events.
+        The return value decides whether to finalize the lib.stroke.Stroke
+        which is currently recording the user's input, and begin recording a
+        new one.
+        """
+        self._surface.begin_atomic()
+        split = brush.stroke_to(
+            self._surface.backend, x, y,
+            pressure, xtilt, ytilt, dtime, viewzoom, viewrotation
+        )
+        self._surface.end_atomic()
+        self.autosave_dirty = True
+        return split
```

Comment:
Looks like like thee additions crept into the patch by accident during a rebase. These methods got moved into a new base class in master.